7.12. Код ревью домашнего задания

GitEsense opened the pull request.

GitEsense committed:
5.19. Домашнее задание - Классы

GitEsense committed:
5.19. Домашнее задание - Классы - v.2

Внесены некоторые исправления, была ошибка логики получения и записи

GitEsense committed:
7.11. Домашнее задание - Generics

GitEsense committed:
7.11. Домашнее задание - Generics

Добавлен .gitignore

SergeyKrasnolobov commented on exercises/5-map/index.ts:
```diff
@@ -0,0 +1,94 @@
+class ManualMap {
+    private _buckets: IBucket[] = [];
+
+    private stringToHash(key: string) {
```
https://ru.wikipedia.org/wiki/%D0%A5%D0%B5%D1%88-%D1%84%D1%83%D0%BD%D0%BA%D1%86%D0%B8%D1%8F

че то мне твоя вариация хеширования не нравится. Что вижу я: для строк 'собака' и 'макака' хеш будет одинаковый, в результате мы получаем коллизию, при этом заполнение бакета при таком хешировании будет происходит неравномерно

SergeyKrasnolobov replied:
советую почитать про хеш таблицы в книге Роберт Лафоре "Структуры данных и алгоритмы в Java"

GitEsense committed:
5.19. Домашнее задание - Классы - v.3

Обновлена и добавлена hash функция, добавлена генерация данных для тестирования
